addons/website_membership/controllers/main.py

```diff
@@ -62,25 +62,27 @@ def members(self, membership_id=None, country_name=None, country_id=0, page=1, *
         if post_name:
             country_domain += ['|', ('name', 'ilike', post_name), ('website_description', 'ilike', post_name)]
 
-        countries = Partner.sudo().read_group(country_domain + [("website_published", "=", True)], ["__count"], groupby="country_id")
-        countries_total = sum(country_dict['country_id_count'] for country_dict in countries)
+        country_groups = Partner.sudo()._read_group(
+            country_domain + [("website_published", "=", True)],
+            ["country_id"], ["__count"], order="country_id")
 
         line_domain = list(base_line_domain)
         if country_id:
             line_domain.append(('partner.country_id', '=', country_id))
-            current_country = Country.browse(country_id).read(['id', 'name'])[0]
-            if not any(x['country_id'][0] == country_id for x in countries if x['country_id']):
-                countries.append({
-                    'country_id_count': 0,
-                    'country_id': (country_id, current_country["name"])
-                })
-                countries = [d for d in countries if d['country_id']]
-                countries.sort(key=lambda d: d['country_id'][1])
-
-        countries.insert(0, {
-            'country_id_count': countries_total,
-            'country_id': (0, _("All Countries"))
-        })
+            if not any(country.id == country_id for country, __ in country_groups):
+                country_groups = [(g_country, count) for g_country, count in country_groups if g_country]
+                country_groups.append((Country.browse(country_id).sudo(), 0))
+                country_groups = sorted(country_groups, key=lambda c, __: c.name or '')
+
+        countries = [{
+            'country_id_count': sum(count for __, count in country_groups),
+            'country_id': (0, _("All Countries")),
+        }]
+        for g_country, count in country_groups:
```

Comment:
Nitpicking: you can make a single comprehension for `countries = [{}, *({} for ...)]`
Same in other website modules.

Reply:
I tried but IMO, it less readable (really just an opinion).